tool/log.re

```diff
@@ -0,0 +1,49 @@
+static mut LOG = Log {
+    messages: List::new(),
+};
+
+struct Log {
+    messages: List<LogMessage>,
+}
+
+struct LogMessage {
+    content: string,
+    added_timestamp: int,
+}
+
+fn log(content: string) {
+    let message = LogMessage {
+        content: content,
+        added_timestamp: current_time_millis(),
+    };
+    LOG.messages.push(message);
+}
+fn clean_expired_messages() {
+    let millis = current_time_millis();
+    let mut i = 0;
+    while i < LOG.messages.len() {
+        if (millis - LOG.messages.get(i).unwrap().added_timestamp) > SETTINGS.log_message_duration {
+            LOG.messages.remove(i);
+        } else {
+            i += 1;
+        }
+    }
+}
+
+fn draw_log_messages() {
+    let viewport = Tas::get_viewport_size();
+    let mut messages = "";
+    for message in LOG.messages {
+        messages = f"{message.content}\n{messages}";
+    }
+    let text_size = Tas::get_text_size(messages, SETTINGS.ui_scale);
```

Comment:
We both now know that the function ignores newlines. However, not everyone will, including us in a few months time. Could you please add a comment in that regard? Could you also add a variable `line_height` to make the code more readable?